pkg/docker/config/config.go

```diff
@@ -0,0 +1,271 @@
+	}
+	if auth != "" {
+		return decodeDockerAuth(auth)
+	}
+
+	// check .docker/config.json as secondary option
+	path = filepath.Join(homedir.Get(), dockerCfg, dockerCfgFileName)
+	auth, err = findAuthentication(registry, path)
+	if err != nil {
+		return "", "", err
+	}
+	if auth != "" {
+		decodeDockerAuth(auth)
+	}
+
+	return "", "", nil
+}
+
+// RemoveAuthentication deletes the credentials stored in auth.json
+func RemoveAuthentication(registry string) (string, error) {
+	path, err := getPathToAuth()
+	if err != nil {
+		return "", errors.Wrapf(err, "error getting path to auth file")
+	}
+
+	auths, err := readJSONFile(path)
+	if err != nil {
+		return "", errors.Wrapf(err, "error reading JSON file %q", path)
+	}
+	if _, ok := auths.AuthConfigs[registry]; ok {
+		delete(auths.AuthConfigs, registry)
+	} else if _, ok := auths.AuthConfigs[normalizeRegistry(registry)]; ok {
+		delete(auths.AuthConfigs, normalizeRegistry(registry))
+	} else {
+		return "", ErrNotLoggedIn
+	}
+
+	file, err := os.OpenFile(path, os.O_RDWR, 0755)
+	if err != nil {
+		return "", errors.Wrapf(err, "error opening file %q", path)
+	}
+	defer file.Close()
+
+	newData, err := json.MarshalIndent(auths, "", "\t")
+	if err != nil {
+		return "", errors.Wrapf(err, "error marshaling JSON %q", path)
+	}
+
+	if err = ioutil.WriteFile(path, newData, 0755); err != nil {
+		return "", errors.Wrapf(err, "error writing to file %q", path)
+	}
+	return registry, nil
+}
+
+// getPath gets the path of the auth.json file since the user can set their own path
+// by setting CRIO_AUTH_PATH
+// If CRIO_AUTH_PATH is not set, the default path is ~/.config/containers/auth.json
+func getPathToAuth() (string, error) {
+	if os.Getenv("CRIO_AUTH_PATH") != "" {
+		return filepath.Join(os.Getenv("CRIO_AUTH_PATH"), authCfgFileName), nil
+	}
+	dir := filepath.Join(homedir.Get(), authCfg)
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		if err = os.Mkdir(dir, 0600); err != nil {
+			return "", errors.Wrapf(err, "error creating directory %q", dir)
+		}
+	}
+	return filepath.Join(dir, authCfgFileName), nil
+}
+
+// readJSONFile unmarshals the authentications stored in the auth.json file
+func readJSONFile(path string) (dockerConfigFile, error) {
+	file, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE, 0755)
+	if err != nil {
+		return dockerConfigFile{}, errors.Wrapf(err, "error creating/opening file %q", path)
+	}
+	defer file.Close()
+
+	var auths dockerConfigFile
+	fStat, err := os.Stat(path)
+	if err != nil {
+		return dockerConfigFile{}, errors.Wrapf(err, "error getting stat of file %q", path)
+	}
+	if fStat.Size() > 0 {
+		raw, err := ioutil.ReadFile(path)
+		if err != nil {
+			return dockerConfigFile{}, errors.Wrapf(err, "error reading file %q", path)
+		}
+
+		if err = json.Unmarshal(raw, &auths); err != nil {
+			return dockerConfigFile{}, errors.Wrapf(err, "error unmarshaling JSON at %q", path)
+		}
+	} else {
+		newMap := make(map[string]dockerAuthConfig)
+		auths.AuthConfigs = newMap
```

Comment:
This could be shorter as `auth.AuthConfigs = map[string]dockerAuthConfig{}`

Reply:
fixed